clients/src/main/java/org/apache/kafka/clients/consumer/internals/LegacyKafkaConsumer.java

```diff
@@ -495,6 +496,16 @@ public void subscribe(Pattern pattern) {
         subscribeInternal(pattern, Optional.empty());
     }
 
+    @Override
+    public void subscribe(SubscriptionPattern pattern, ConsumerRebalanceListener callback) {
+        throw new IllegalArgumentException("Operation not supported in the classic group protocol");
```

Comment:
Throwing an `IllegalArgumentException` seems just a tiny bit weird. What about `UnsupportedOperationException` instead?

Reply:
There was some discussion about implementing these APIs in the `ClassicKafkaConsumer`. IIUC, re2j patterns are a strict subset of those used by the bundled Java libraries in `java.util.regex`. CMIIW, though, @AndrewJSchofield.